crates/lsp/src/lsp.rs

```diff
@@ -1229,13 +1287,24 @@ impl LanguageServer {
     ///
     /// [LSP Specification](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#notificationMessage)
     pub fn notify<T: notification::Notification>(&self, params: &T::Params) -> Result<()> {
-        Self::notify_internal::<T>(&self.outbound_tx, params)
+        Self::notify_internal::<T>(&self.outbound_tx, self.server_id, params)
     }
 
     fn notify_internal<T: notification::Notification>(
         outbound_tx: &channel::Sender<String>,
+        server_id: LanguageServerId,
         params: &T::Params,
     ) -> Result<()> {
+        // Log outgoing notification with params
+        let params_str =
+            serde_json::to_string_pretty(&params).unwrap_or_else(|_| "...".to_string());
+        log::info!(
```

Comment:
`log::info` that is enabled by default and prints all LSP JSON requests into the file???????

This is an insane volume of information printed into a file.
Here's what I get for a small project out of a single language server's startup logs:

https://github.com/user-attachments/assets/61207bdd-036b-4f88-9072-a476b663ded8